homework.js

```diff
@@ -0,0 +1,212 @@
+//Q1
+
+const  summation = function (n) {
+  let output = 0;
+  for( i = 1;  i <= n; i++) {
+    output = output  + i;
+  }
+  return output ;
+};
+
+let n = 5 ;
+console.log('Sum of numebers from 1 to '+ n +' is '+ summation(n));
+
+
+//Q2
+
+const summationEven = function (n){
+  let sum = 0 ;
+ 
+ for ( i=0; i <=n; i=i+2) {
+   sum +=1
+   }
+ return sum ;
+ 
+ };
+  let n = 6
+ console.log(summationEven(n));
+
+
+ //Q3
+
+ const avg=[8, 4, 2, 4];
+
+const getAvg = function () {
+
+for (i = 0 ; i< avg.length ;i++) 
+{
+ if (avg[i] === avg.length ){
+   
+ avg= avg[i] + avg[i]; 
+ }
+return avg[i] / avg.length ;
+}
+};
+
+console.log(getAvg())
+
+//Q4 
+
+function reverseString(str) {
+var splitString = str.split("")
+
+
+var reverseArray = splitString.reverse(); 
+
+
+var joinArray = reverseArray.join("");
+
+return joinArray;
+}
+
+console.log(reverseString("caterpillar"));
+
+//Q5
+
+const addDashes =['test1', 'test2', 'test3'] ;
+function combinesThemDash () { 
+  let Dash ;
+for (i=0 ; i <= addDashes.length ; i++){
+if (i === addDashes.length)
+Dash = addDashes.join("-")
+}
+return Dash  ;
+};
+console.log(combinesThemDash());
+
+or 
+
+const addDashes= function(Dasge){
+
+return Dasge.join("-");
+
+}
+
+console.log(addDashes(['test1', 'test2', 'test3']));
+
+
+
+//Q6
+
+function printNumbers(args) {
+  let input = args[0];
+  for (var i = 1; i <= input; i++) {
+    console.log(i);
+    //return i;
+  }
+  for (var j = input-1; j <= input; j--) {
+    console.log(j);
+    if (j<=1){
+      break;
+    }
+  }
+}
+console.log(printNumbers("3"));
+
+//Q7
+
+const LetterA = ['cat', 'rabbit', 'dog', 'frog']
+const FindA = 'a';
+function find(LetterA,FindA){
```

Comment:
not the desired input, the output is great but function needs to be restructured to match the requirements